examples/templates/api/streaming-file-to-file.md

```diff
@@ -0,0 +1,138 @@
+[//]: # "Copyright Amazon.com Inc. or its affiliates. All Rights Reserved."
+[//]: # "SPDX-License-Identifier: CC-BY-SA-4.0"
+
+# Basic use of stream APIs example using files
+
+Implementations of this example MUST follow the rules defined in
+[Example Templates](../../../examples.md#example-templates).
+
+## Implementations
+
+- [Python (DEV)](https://github.com/aws/aws-encryption-sdk-python/blob/keyring/examples/src/file_streaming_defaults.py)
+- [Java (DEV)](https://github.com/aws/aws-encryption-sdk-java/blob/keyring/src/examples/java/com/amazonaws/crypto/examples/FileStreamingDefaults.java)
+
+## Definitions
+
+### Conventions used in this document
+
+The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in
+[RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Header
+
+```python
+# This example shows how to use the streaming encrypt and decrypt APIs when working with files.
+#
+# One benefit of using the streaming API is that
+# we can check the encryption context in the header before we start decrypting.
+#
+# In this example, we use an AWS KMS customer master key (CMK),
+# but you can use other key management options with the AWS Encryption SDK.
+# For examples that demonstrate how to use other key management configurations,
+# see the ``keyring`` and ``master_key_provider`` directories.
+```
+
+## Summary
+
+```python
+# Demonstrate an encrypt/decrypt cycle using the streaming encrypt/decrypt APIs with files.
+```
+
+## Inputs
+
+- **source plaintext filename** :
+  Path to plaintext file to encrypt
+- **AWS KMS CMK** :
+  The ARN of an AWS KMS CMK that protects data keys
+
+## Steps
+
+1. Define file targets.
+
+   ```python
+   # We assume that you can also write to the directory containing the plaintext file,
+   # so that is where we will put all of the results.
+   ciphertext_filename = source_plaintext_filename + ".encrypted"
+   decrypted_filename = ciphertext_filename + ".decrypted"
+   ```
+
+1. Define encryption context.
+
+   ```python
+   # Prepare your encryption context.
+   # Remember that your encryption context is NOT SECRET.
+   # https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/concepts.html#encryption-context
+   encryption_context = {
+       "encryption": "context",
+       "is not": "secret",
+       "but adds": "useful metadata",
+       "that can help you": "be confident that",
+       "the data you are handling": "is what you think it is",
+   }
+   ```
+
+1. Create keyring.
+
+   ```python
+   # Create the keyring that determines how your data keys are protected.
+   keyring = AwsKmsKeyring(generator_key_id=aws_kms_cmk)
+   ```
+
+1. Encrypt plaintext data.
+
+   ```python
+   # Open the files you want to work with.
+   with open(source_plaintext_filename, "rb") as plaintext, open(ciphertext_filename, "wb") as ciphertext:
```

Comment:
Yup, I think that this is so language specific,
that maybe nothing is better?
(in the long run :) no need to block )